lib/miq_ldap.rb

```diff
@@ -266,16 +267,30 @@ def normalize(dn)
     dn.split(",").collect { |i| i.downcase.strip }.join(",")
   end
 
+  def self.dn?(str)
+    Net::LDAP::DN.new(str).to_a.any?
+  rescue Net::LDAP::Error
+    false
+  end
+
   def dn?(str)
-    !!(str =~ /^([a-z|0-9|A-Z]+ *=[^,]+[,| ]*)+$/)
+    self.class.dn?(str)
+  end
+
+  def self.upn?(str)
+    str.to_s.include?("@")
   end
 
   def upn?(str)
-    !!(str =~ /^.+@.+$/)
```

Comment:
Note this just checks for a '@'.  _Technically_ it checks for at least one char before and one char after the '@', so it's not completely accurate, so I can change to something like `/.@./` if that's deemed better.

Reply:
Update: I changed it to go with the first example out of simplicity.  If you like the other ones better, let me know.